syntax = "proto3";

package mcp;

service MCPService {
    rpc GetRelevantTools(GetToolsRequest) returns (GetToolsResponse);
    rpc ExecuteTool(ExecuteToolRequest) returns (ExecuteToolResponse);
    rpc GetPageElements(GetPageElementsRequest) returns (GetPageElementsResponse);
}

message GetToolsRequest {
    string user_id = 1;
    string query = 2;
    string auth_token = 3;
}

message Tool {
    string name = 1;
    string description = 2;
    string parameters_schema = 3;
}

message GetToolsResponse {
    bool success = 1;
    repeated Tool tools = 2;
    string context = 3;
}

message ExecuteToolRequest {
    string user_id = 1;
    string tool_name = 2;
    string arguments_json = 3;
    string auth_token = 4;
}

message ExecuteToolResponse {
    bool success = 1;
    string result = 2;
    string error = 3;
}

message GetPageElementsRequest {
    string user_id = 1;
    string auth_token = 2;
}

message UIElement {
    string id = 1;
    string type = 2;
    string description = 3;
    string text = 4;
    string context = 5;
}

message GetPageElementsResponse {
    bool success = 1;
    repeated UIElement elements = 2;
    string current_url = 3;
    string page_title = 4;
    string error = 5;
}
